src/modules/75-cf/pages/feature-flags-detail/flag-pipeline-tab/__tests__/FlagPipelineTab.test.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import { render, screen, waitFor } from '@testing-library/react'
+import React from 'react'
+import userEvent from '@testing-library/user-event'
+import { TestWrapper } from '@common/utils/testUtils'
+import * as cfServicesMock from 'services/cf'
+import FlagPipelineTab, { FlagPipelineTabProps } from '../FlagPipelineTab'
+import mockAvailablePipelines from './__data__/mockAvailablePipelines'
+
+const renderComponent = (props: Partial<FlagPipelineTabProps> = {}): void => {
+  render(
+    <TestWrapper
+      path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/feature-flags/TEST_FLAG"
+      pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
+      queryParams={{ tab: 'flag_pipeline', activeEnvironment: 'TEST_ENV' }}
+    >
+      <FlagPipelineTab flagIdentifier="TEST_FLAG" {...props} />
+    </TestWrapper>
+  )
+}
+
+describe('FlagPipelineTab', () => {
+  const refetchMock = jest.fn()
+  beforeAll(() => {
+    jest.spyOn(cfServicesMock, 'useGetAvailableFeaturePipelines').mockReturnValue({
+      data: {
+        availablePipelines: mockAvailablePipelines
+      },
+      loading: false,
+      refetch: refetchMock
+    } as any)
+  })
+
+  test('it should let user select a flag pipeline when non configured', async () => {
+    renderComponent()
+
+    // assert empty state initially
+    const addFlagPipelineButton = screen.getByRole('button', { name: 'cf.featureFlags.flagPipeline.noDataButtonText' })
+    expect(screen.getByText('cf.featureFlags.flagPipeline.noDataMessage')).toBeInTheDocument()
+    expect(addFlagPipelineButton).toBeInTheDocument()
+    expect(screen.getByText('cf.featureFlags.flagPipeline.noDataDescription')).toBeInTheDocument()
+
+    // assert drawer opens with correct text. Save button hidden initially
+    addFlagPipelineButton.click()
+    expect(screen.getByText('cf.featureFlags.flagPipeline.drawerTitle')).toBeInTheDocument()
+    expect(screen.getByText('cf.featureFlags.flagPipeline.drawerDescription')).toBeInTheDocument()
+    expect(screen.getByText('cf.featureFlags.flagPipeline.drawerEnvText')).toBeInTheDocument()
+    expect(
+      screen.queryByRole('button', { name: 'cf.featureFlags.flagPipeline.drawerButtonText' })
+    ).not.toBeInTheDocument()
+
+    // assert correct number of pipelines and pipeline content is displayed
+    await waitFor(() => expect(screen.getAllByRole('listitem')).toHaveLength(mockAvailablePipelines.length))
+    await waitFor(() =>
+      expect(screen.getAllByRole('listitem')[0]).toHaveTextContent('Pipeline 1Pipeline 1 description')
+    )
```

Comment:
You can do this in a single `waitFor`
```suggestion
    await waitFor(() => {
      expect(screen.getAllByRole('listitem')).toHaveLength(mockAvailablePipelines.length))
      expect(screen.getAllByRole('listitem')[0]).toHaveTextContent('Pipeline 1Pipeline 1 description')
    })
```

Reply:
will fix